src/opossum_lib/file_generation.py

```diff
@@ -4,21 +4,40 @@
 import datetime
 import json
 from pathlib import Path
-from typing import Any, Dict, Union
+from typing import Any, Dict, List, TypedDict, Union
 
-from networkx import DiGraph, weakly_connected_components
+from networkx import DiGraph, shortest_path, weakly_connected_components
 
-from opossum_lib.helper_methods import _get_source_for_graph_traversal
+from opossum_lib.helper_methods import (
+    _get_source_for_graph_traversal,
+    _node_represents_a_spdx_element,
+)
 
+OpossumInformation = TypedDict(
+    "OpossumInformation",
+    {
+        "metadata": Dict[str, str],
+        "resources": Dict[str, Any],
+        "externalAttributions": Dict,
+        "resourcesToAttributions": Dict,
+        "attributionBreakpoints": List[str],
+    },
+)
```

Comment:
Is there a special reason you are using this functional syntax to define the typed dict? Looking at https://docs.python.org/3/library/typing.html#typing.TypedDict it seems to me that the other syntax is more common.

Reply:
No special reason, I didn't work with `TypedDict` before and iirc this was just the result of my google search on a mypy error I had. I will change this to use the more common syntax.